src/httpserver.cpp

```diff
@@ -146,6 +148,10 @@ static GlobalMutex g_httppathhandlers_mutex;
 static std::vector<HTTPPathHandler> pathHandlers GUARDED_BY(g_httppathhandlers_mutex);
 //! Bound listening sockets
 static std::vector<evhttp_bound_socket *> boundSockets;
+//! Track active requests
+static GlobalMutex g_requests_mutex;
+static std::condition_variable g_requests_cv;
+static std::set<evhttp_request*> g_requests GUARDED_BY(g_requests_mutex);
```

Comment:
Is any particular reason for not using `std::unordered_set` which has more efficient `insert()` and `erase()` methods?

Reply:
No reason, should be a small set anyway, both are fine IMHO.

Reply:
done